internal/pkg/chainsync/internal/dispatcher/dispatcher.go

```diff
@@ -205,6 +206,19 @@ func (d *Dispatcher) RegisterCallback(cb func(Target)) {
 	d.control <- cbMessage{cb: cb}
 }
 
+// WaitForTarget returns a function that will block until the dispatcher
+// processes the given target
+func (d *Dispatcher) WaitForTarget(ci *block.ChainInfo) func() {
```

Comment:
The parameter should be just the tipset key `ci.Head` since other fields are ignored.